diff --git a/binding.gyp b/binding.gyp
index 6d3bf338bb6c0ca8fa10dc92a0b831e9f7ec25dd..756e58e0f60e4bd2dc0e053144ed3b6775466a2a 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -37,16 +37,20 @@
 		{
 			"target_name": "action_after_build",
 			"type": "none",
-			"dependencies": [
-				"<(module_name)",
-			],
-			"copies": [
-				{
-					"files": [
-						"<(PRODUCT_DIR)/<(module_name).node",
+			"conditions": [
+				["OS=='win'", {
+					"dependencies": [
+						"<(module_name)",
 					],
-					"destination": "<(module_path)"
-				}
+					"copies": [
+						{
+							"files": [
+								"<(PRODUCT_DIR)/<(module_name).node",
+							],
+							"destination": "<(module_path)"
+						}
+					]
+				}]
 			]
 		}
 	]
diff --git a/lib/windows.js b/lib/windows.js
index b040c7401b79c90017cfbeecdf3d1a792e281ed3..5734bb23b257ed5eded753c1d8937bffe2216816 100644
--- a/lib/windows.js
+++ b/lib/windows.js
@@ -10,10 +10,15 @@ const require = createRequire(import.meta.url);
 
 const bindingPath = preGyp.find(path.resolve(path.join(__dirname, '../package.json')));
 
-const addon = (fs.existsSync(bindingPath)) ? require(bindingPath) : {
+let addon = {
 	getActiveWindow() {},
 	getOpenWindows() {},
 };
+try {
+	if (fs.existsSync(bindingPath)) {
+		addon = require(bindingPath);
+	}
+} catch (err) {}
 
 export async function activeWindow() {
 	return addon.getActiveWindow();
diff --git a/package.json b/package.json
index a41f2507ad0038e89219ee4eeb0f5d107037d344..091f77bea925be8856c238d136641bcfee4ceade 100644
--- a/package.json
+++ b/package.json
@@ -40,7 +40,7 @@
 		"build:windows": "node-pre-gyp build",
 		"build:windows:debug": "node-pre-gyp build --debug",
 		"build:macos": "swift build --configuration=release --arch arm64 --arch x86_64 && mv .build/apple/Products/Release/active-win main",
-		"install": "node-pre-gyp install --fallback-to-build"
+		"install": "node -e \"if (process.platform === 'win32') { const cp = require('child_process'); const res = cp.spawnSync('node-pre-gyp', ['install', '--fallback-to-build'], { stdio: 'inherit', shell: true }); process.exit(res.status); } else { process.exit(0); }\""
 	},
 	"files": [
 		"index.js",
